### ISSUE TITLE
not starting on "Pocketbook Inkpad 3 Pro"

### ISSUE THREAD

**Issue by tobiasarndt** [koreader/koreader#5397]
* KOReader version: v2019.09.3
* Device: Pocketbook Inkpad 3 Pro

#### Issue

Crashing upon start, because there is no entry for the device in frontend\device\pocketbook\device.lua

#### Steps to reproduce

Starting koreader

##### `crash.log` (if applicable)
./luajit: frontend/device/pocketbook/device.lua:365: unrecognized PocketBook model PB740-2

#### Workaround
Because the Inkpad 3 and the Inkpad 3 Pro are nearly identical, this issue can be worked around
by adding
```
elseif codename == "PB740-2" then
    return PocketBook740
```
in the device.lua

**Comment by Frenzie** [comment]
By nearly identical, do you also mean the DPI? If so, please submit a PR! :-)

**Comment by tobiasarndt** [comment]
> By nearly identical, do you also mean the DPI? If so, please submit a PR! :-)

As far as i know the devices have the exact same hardware and firmware, except for the "Pro" version having a Bluetooth-Module and Waterproofing. I didn't know whether to make a pull request, because the workaround seemed kinda hacky.

I just tried creating a PR, but being a github newbie i can't figure out how to make it work.

**Comment by Frenzie** [comment]
If the hardware is literally the same then I wouldn't call that hacky. I would add a little comment about the Bluetooth though, and perhaps it's clearer to put that in the following position instead.

```diff
diff --git a/frontend/device/pocketbook/device.lua b/frontend/device/pocketbook/device.lua
index 148130c0..eab6c344 100644
--- a/frontend/device/pocketbook/device.lua
+++ b/frontend/device/pocketbook/device.lua
@@ -297,6 +297,10 @@ local PocketBook740 = PocketBook:new{
     emu_events_dev = "/var/dev/shm/emu_events",
 }
 
+-- PocketBook InkPad 3 Pro
+-- Same as the PB InkPad 3, but with Bluetooth
+local PocketBook740_2 = PocketBook740
+
 -- PocketBook Sense
 local PocketBook630 = PocketBook:new{
     model = "PBSense",
```

To create PR there are two ways. The easiest way for something small like this is just to click edit on the file in question (top right), put your changes in place and to submit them as a PR.

[This link](https://github.com/koreader/koreader/edit/master/frontend/device/pocketbook/device.lua) might work.

**Comment by tobiasarndt** [comment]
Thanks a lot. Just put in the PR. In order to be consistent with the rest of the file i made a new object for the Pro.

```
-- PocketBook InkPad 3 Pro
local PocketBook740_2 = PocketBook:new{
    model = "PBInkPad3Pro",
    isTouchDevice = yes,
    hasKeys = yes,
    hasFrontlight = yes,
    isAlwaysPortrait = yes,
    display_dpi = 300,
    emu_events_dev = "/var/dev/shm/emu_events",
}
```

**Comment by Frenzie** [comment]
Closed by #5398.
